PatchesAndAreasApi.Tests/V1/Controllers/PatchesAndAreasApiControllerTests.cs

```diff
@@ -327,5 +330,63 @@ public async void GetAllPatches_ReturnsOkResultWithPatches()
             var outcome = Assert.IsType<OkObjectResult>(result);
             var resultPatches = Assert.IsType<List<PatchesResponseObject>>(outcome.Value);
         }
+        [Fact]
+        public async Task ReplacePatchResponsibleEntitiesAsyncFoundReturnsFound()
+        {
+            // Arrange
+            var query = ConstructQuery();
+            var request = _fixture.Build<ResponsibleEntities>().CreateMany(2).ToList();
+            var patchResponse = _fixture.Create<PatchesResponseObject>();
+            _mockReplacePatchResponsibleEntitiesUseCase.Setup(x => x.ExecuteAsync(query, request, It.IsAny<int?>()))
+                                    .ReturnsAsync(patchResponse);
+
+            // Act
+            var response = await _classUnderTest.ReplacePatchResponsibleEntities(query, request).ConfigureAwait(false);
+
+            // Assert
+            response.Should().BeOfType(typeof(NoContentResult));
+        }
+
+        [Fact]
+        public async Task ReplacePatchResponsibleEntitiesNotFoundReturnsNotFound()
+        {
+            // Arrange
+            var query = ConstructQuery();
+            var request = _fixture.Build<ResponsibleEntities>().CreateMany(2).ToList();
+            var patchResponse = _fixture.Create<PatchesResponseObject>();
+            _mockReplacePatchResponsibleEntitiesUseCase.Setup(x => x.ExecuteAsync(query, request, It.IsAny<int?>()))
+                                    .ReturnsAsync((PatchesResponseObject) null);
+
+            // Act
+            var response = await _classUnderTest.ReplacePatchResponsibleEntities(query, request).ConfigureAwait(false);
+
+            // Assert
+            response.Should().BeOfType(typeof(NotFoundObjectResult));
+            (response as NotFoundObjectResult).Value.Should().Be(query.Id);
+        }
+
+        [Theory]
+        [InlineData(null, 0)]
+        [InlineData(0, 1)]
+        [InlineData(0, null)]
+        [InlineData(2, 1)]
+        public async Task ReplacePatchResponsibleEntitiesAsyncVersionNumberConflictExceptionReturns409(int? expected, int? actual)
```

Comment:
Should a missing version number be a bad request?

Reply:
Missing version number returns a 409 Conflict and then the response has a message that guides the user to add a version number to the header with the current version number, here is an example:
`"The version number supplied ({null}) does not match the current value on the entity (2)."`